llvm/lib/Target/AMDGPU/SIInstrInfo.td

```diff
@@ -284,12 +284,17 @@ def SIfptrunc_round_downward : SDNode<"AMDGPUISD::FPTRUNC_ROUND_DOWNWARD",
 // Returns 1 if the source arguments have modifiers, 0 if they do not.
 class isFloatType<ValueType SrcVT> {
   bit ret = !or(!eq(SrcVT.Value, f16.Value),
```

Comment:
ValueType has an isFP field; we could probably get rid of isFloatType altogether

Reply:
Also isVector and isInteger

Reply:
https://github.com/llvm/llvm-project/pull/77987